Use Spring IO platform for dependency management

pcvolkmer opened the pull request:
See http://platform.spring.io/platform/ for more information about it.

I overrode pre defined versions for spring-security.

This patch set uses Spring.IO  2.0.0 and also updates some dependiencies not beeing upgraded in ARSnova Backend POM so I removed explicit version declaration.

Problems with json-path-assert dit not occure, so i removed version declaration and use managed version 2.0.0.

dgerhardt commented:
I think we should wait for the release of Spring IO 2.0. It does not make much sense to use it and then override dependency versions. Since Spring 4.2 is used now by ARSnova another dependency would have to be overridden otherwise.

pcvolkmer commented:
I upgraded to Spring.IO 2.0.0

This latest patch makes Spring.IO usable with ARSnova Backend and also updates some dependiencies not beeing upgraded in ARSnova Backend POM so I removed explicit version declaration.

Problems with json-path-assert dit not occure, so i removed version
declaration and use managed version 2.0.0.

dgerhardt commented:
Thank you for your continuous work on this PR. I'll have a closer look on this shortly.

Could you please squash your PR's commits into a single one to ease following of changes in our repo?
